src/Sentry.Unity/SentryUnityOptions.cs

```diff
@@ -111,6 +112,26 @@ public new CompressionLevelWithAuto RequestBodyCompressionLevel
         /// </summary>
         public bool LinuxNativeSupportEnabled { get; set; } = true;
 
+
+        // Initialized by native SDK binding code to set the User.ID in .NET (UnityEventProcessor).
+        internal string? _defaultUserId;
+        internal string? DefaultUserId
+        {
+            get => _defaultUserId;
+            set
+            {
+                _defaultUserId = value;
+                if (_defaultUserId is null)
+                {
+                    DiagnosticLogger?.LogWarning("Couldn't set the default user ID - the value is NULL.");
+                }
+                else
+                {
+                    DiagnosticLogger?.LogDebug("Setting '{0}' as the default user ID.", _defaultUserId);
+                }
```

Comment:
```suggestion
                if (value is null)
                {
                    DiagnosticLogger?.LogWarning("Couldn't set the default user ID - the value is NULL.");
                }
                else
                {
                    _defaultUserId = value;
                    DiagnosticLogger?.LogDebug("Setting '{0}' as the default user ID.", _defaultUserId);
                }
```
Should this not set the id if it logs that it can't?

Reply:
Is the goal to reset to null as it'll be needed later to "unset"?

Reply:
> Should this not set the id if it logs that it can't?

this is a setter - doesn't feel right to ignore the value being passed (i.e. not set)